src/OrchardCore/Orchard.Environment.Shell/ShellSettingsConfigurationProvider.cs

```diff
@@ -50,23 +50,14 @@ public void SaveToSource(string name, IDictionary<string, string> configuration)
                 Optional = false
             });
 
-            configurationProvider.Set(name, null);
-            configurationProvider.Set($"{name}:RequestUrlHost", ObtainValue(configuration,$"{name}:RequestUrlHost"));
-            configurationProvider.Set($"{name}:RequestUrlPrefix", ObtainValue(configuration,$"{name}:RequestUrlPrefix"));
-            configurationProvider.Set($"{name}:DatabaseProvider", ObtainValue(configuration,$"{name}:DatabaseProvider"));
-            configurationProvider.Set($"{name}:TablePrefix", ObtainValue(configuration,$"{name}:TablePrefix"));
-            configurationProvider.Set($"{name}:ConnectionString", ObtainValue(configuration,$"{name}:ConnectionString"));
-            configurationProvider.Set($"{name}:State", ObtainValue(configuration,$"{name}:State"));
+            foreach (var item in configuration)
+            {
+                configurationProvider.Set($"{item.Key}", item.Value ?? "~");
```

Comment:
I must be misunderstanding how this works then. I thought that this was just the site settings serialised. I'll revert this change and resubmit. 

Are you happy with using "~" as null though as it makes people check the spec before raising a bug like I did.